decidim-core/app/services/decidim/resource_search.rb

```diff
@@ -39,12 +39,22 @@ def search_category_id
     # `"global"` as parameter, and in the method we do the needed changes to search
     # properly.
     #
-    # You can use the `search_organization_scopes` helper method, defined in
-    # `Decidim::OrganizationScopesHelper`, to render the collection needed for the
+    # You can use the `search_participatory_process_subscopes` helper method, defined in
+    # `Decidim::ParticipatoryProcessScopesHelper`, to render the collection needed for the
     # `collection_check_boxes` form method.
     def search_scope_id
-      clean_scope_ids = [scope_id].flatten.map { |id| id == "global" ? nil : id }
-      query.where(decidim_scope_id: clean_scope_ids)
+      clean_scope_ids = [scope_id].flatten
+
+      conditions = []
+      if clean_scope_ids.delete("global")
+        conditions << "decidim_scope_id IS NULL"
+      end
+      if clean_scope_ids.any?
+        self.query = query.joins("LEFT JOIN decidim_scopes ON decidim_scope_id=decidim_scopes.id")
+        conditions.concat ["? = ANY(decidim_scopes.part_of)"] * clean_scope_ids.count
```

Comment:
Not sure about this specific feature since I haven't looked at it closely. But many times the right solution ends up being the simpler one too. It would certainly simplify the bit of code... :)